src/discord/events/message/supportRequest.ts

```diff
@@ -1,32 +1,78 @@
 import Discord from 'discord.js';
-import { SupportRequest, SupportRequestType, SupportRequestErrors } from '../../../entities/supportRequest';
+import { Not } from 'typeorm';
+import { SupportRequest, SupportRequestType, SupportRequestStatus } from '../../../entities/supportRequest';
 import logger from '../../../logger';
 import { Config } from '../../../entities/config';
+import { SubCommands } from '.';
+import { DiscordContext } from '../../../entities/discordContext';
 
-export async function supportRequest(msg: Discord.Message): Promise<void> {
-  const discordId = msg.author.id;
-  const discordName = msg.author.username;
-  const actionId = msg.content === '!technicalSupport' ? SupportRequestType.TechnicalSupport : SupportRequestType.IdeaPitch;
+/* eslint-disable no-param-reassign */
 
+interface UserInfo {
+  [k: string]: number | string | string[];
+  id: string;
+  requestType: SupportRequestType;
+  username: string;
+}
+
+enum steps {
+  inputName = 'inputName',
+}
+
+const requestTypeMapping: { [id: string]: SupportRequestType } = {
+  '!technicalSupport': SupportRequestType.TechnicalSupport,
+  '!ideaPitch': SupportRequestType.IdeaPitch,
+  '!jobChat': SupportRequestType.JobChat,
+};
+
+export async function supportRequest(msg: Discord.Message, context: DiscordContext): Promise<void> {
   const supportRequestQueueActive = await Config.findToggleForKey('supportRequestQueueActive');
   if (!supportRequestQueueActive) {
     msg.author.send("**Whoops...**\n:see_no_evil: Our team isn't available to help at the moment, check back with us soon!");
-  } else {
+    return;
+  }
+
+  const userHasOpenRequests =
+    (await SupportRequest.createQueryBuilder()
+      .where({ slackId: msg.author.id, status: Not(SupportRequestStatus.Complete) })
+      .getCount()) > 0;
+  if (userHasOpenRequests) {
+    await msg.author.send(
+      "**Whoops...**\n:warning: Looks like you're already waiting to get help from our team\nKeep an eye on your direct messages from this bot for updates. If you think this is an error, come chat with our team.",
+    );
+    context.clear();
+    return;
+  }
+  const payloadInfo: UserInfo = {
+    id: msg.author.id,
+    requestType: requestTypeMapping[msg.content],
+    username: '',
+  };
+  const cmdName = msg.content.replace('!', '');
+  const info = payloadInfo;
+  const prompt = payloadInfo.requestType === SupportRequestType.JobChat ? "what's your name" : "what's the name of your team's voice channel";
+  msg.author.send(`Hey there :wave: before we add you to the queue, ${prompt}?`);
+  context.nextStep = steps.inputName;
+  context.currentCommand = cmdName;
+  context.payload = info;
+  await context.save();
+}
+
+export const supportRequestSubCommands: SubCommands = {
+  inputName: async (msg, ctx) => {
+    const info = ctx.payload as UserInfo;
+    info.username = msg.content;
```

Comment:
This should probably be trimmed in case of any stray spaces

```suggestion
    info.username = msg.content.trim();
```

Reply:
I had that thought too, but I don't think the handler should have to trim the message... 

Here's our current logic:
``` typescript
  handler = commands.find((c) => c.trigger === msg.content.trim())?.handler;
```

We're only checking to see if a matching command can be found for the whole command, meaning we're not checking for content after. What if we just trim the original `msg.content` so other handlers don't need to trim it?

Reply:
Yeah, that sounds good. I can't think of a scenario where we wouldn't want it trimmed. (Ascii art maybe? lol)

Reply:
@Ulfingrad can you modify the line above within the generic handler to remove the `.trim` and instead modify the original? You'll just need to add something like this to the top of the generic handler.
``` typescript
msg.content = msg.content.trim();
```